src/sentry_transport.h

```diff
@@ -5,6 +5,19 @@
 
 typedef struct sentry_rate_limiter_s sentry_rate_limiter_t;
 
+/**
+ * The transport interface of sentry.
+ */
+// TODO: make this completely opaque, and rather use explicit functions from
+// core
```

Comment:
Can we resolve this TODO in this PR already? It would be great to start reducing the number of TODOs.